src/http/gateway/routes/gateway.js

```diff
@@ -2,19 +2,37 @@
 
 const resources = require('../resources')
 
-module.exports = {
-  method: '*',
-  path: '/ipfs/{cid*}',
-  options: {
-    handler: resources.gateway.handler,
-    pre: [
-      { method: resources.gateway.checkCID, assign: 'args' }
-    ],
-    response: {
-      ranges: false // disable built-in support, we do it manually
-    },
-    ext: {
-      onPostHandler: { method: resources.gateway.afterHandler }
+module.exports = [
+  {
+    method: '*',
+    path: '/ipfs/{immutableId*}',
+    options: {
+      handler: resources.gateway.handler,
+      pre: [
+        { method: resources.gateway.checkImmutableId, assign: 'args' }
+      ],
+      response: {
+        ranges: false // disable built-in support, we do it manually
+      },
+      ext: {
+        onPostHandler: { method: resources.gateway.afterHandler }
+      }
+    }
+  },
+  {
+    method: '*',
+    path: '/ipns/{mutableId*}',
```

Comment:
we should define what we should call this id, libp2p-key, ipns-key-id, even ipns-cid (content been the hash of the pubkey) and we should PR to the spec https://github.com/ipfs/specs/tree/master/naming this new name.

Reply:
We have [libp2p-key multicodec now](https://github.com/multiformats/js-multicodec/pull/45) but it also can be FQDN with DNSLink, so I renamed it to `libp2pKeyOrFqdn`